api/endpoint.go

```diff
@@ -57,7 +61,15 @@ func retrieveEndpoint(svc callhome.Service) endpoint.Endpoint {
 
 func retrieveSummaryEndpoint(svc callhome.Service) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
-		summary, err := svc.RetrieveSummary(ctx)
+		req := request.(listTelemetryReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		filter := callhome.TelemetryFilters{
+			From: req.from,
+			To:   req.to,
+		}
+		summary, err := svc.RetrieveSummary(ctx, filter)
```

Comment:
@SammyOina , Return type is not `mainflux.Response`
It is by design ?
Since it is not `mainflux.Response` , content-type is not setted in response header.
I can see other Endpoints are returning as `mainflux.Response` except `serverUI`

Reply:
Good catch I must have missed that